src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -197,11 +202,14 @@ export interface ITerminalEditorService extends ITerminalInstanceHost, ITerminal
 	/** Gets all _terminal editor_ instances. */
 	readonly instances: readonly ITerminalInstance[];
 
+	onDidRequestDetachInstance: Event<{ workspaceId: string, instanceId: number }>;
+
 	openEditor(instance: ITerminalInstance): Promise<void>;
-	getOrCreateEditorInput(instance: ITerminalInstance | SerializedTerminalEditorInput): TerminalEditorInput;
+	getOrCreateEditorInput(instance: ITerminalInstance | SerializedTerminalEditorInput | URI): TerminalEditorInput;
 	detachActiveEditorInstance(): ITerminalInstance;
 	detachInstance(instance: ITerminalInstance): void;
 	splitInstance(instanceToSplit: ITerminalInstance, shellLaunchConfig?: IShellLaunchConfig): ITerminalInstance;
+	parseTerminalUri(resource: URI): ITerminalIdentifier;
```

Comment:
todo: move this elsewhere since it's also used in cases that might be panel terminals?